### PR TITLE
FCC segmentation classes migration to k4geo

### PR DESCRIPTION
**Pull request by giovannimarchiori**
BEGINRELEASENOTES
- Code changes related to k4geo migration of FCC segmentation classes and related utilities
- Also moves HCal readout for Allegro from eta-based to theta-based
- Basically a rebasing of https://github.com/key4hep/k4geo/pull/296 after https://github.com/key4hep/k4geo/pull/298 was merged
ENDRELEASENOTES

### CODE REVIEW

**Comment by BrieucF** [comment]
@andresailer , could you take a look at this PR when possible? Thanks

**Comment by BrieucF** [comment]
Good to be merged?

File: detectorCommon/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+################################################################################
+# Package: detectorCommon
+################################################################################
+
+file(GLOB sources src/*.cpp)
+add_library(detectorCommon SHARED ${sources})
+target_link_libraries(detectorCommon DD4hep::DDCore DD4hep::DDG4 detectorSegmentations)
+target_include_directories(detectorCommon
+    PUBLIC
+        $<INSTALL_INTERFACE:include>
```
[the remark this answers is not in the document]
Comment (giovannimarchiori) [reply detectorCommon/CMakeLists.txt]: fixed in latest commit

File: detectorCommon/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+file(GLOB headers include/detectorCommon/*.h)
+set_target_properties(detectorCommon PROPERTIES PUBLIC_HEADER "${headers}")
+
+install(TARGETS detectorCommon
+  EXPORT ${PROJECT_NAME}Targets
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT bin
+  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT shlib
+  PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_PREFIX}/include/detectorCommon"
```
[the remark this answers is not in the document]
Comment (giovannimarchiori) [reply detectorCommon/CMakeLists.txt]: fixed in latest commit

File: detectorCommon/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  COMPONENT dev)
+
+
+dd4hep_generate_rootmap(detectorCommon)
```
[the remark this answers is not in the document]
Comment (giovannimarchiori) [reply detectorCommon/CMakeLists.txt]: in latest commit I removed dd4hep_generate_rootmap(detectorCommon), let me know if I should also remove COMPONENT dev

[the remark this answers is not in the document]
Comment (andresailer) [reply detectorCommon/CMakeLists.txt]: The install for the headers is still needed.

**Review by andresailer — APPROVED** [review]

**Pull request merged**